management-api-server/src/test/java/com/datastax/mgmtapi/NonDestructiveOpsIT.java

```diff
@@ -1045,12 +1039,35 @@ public void testMoveNode() throws IOException, URISyntaxException {
             });
   }
 
+  @Test
+  public void testGetTokenRangeToEndpointMap() throws Exception {
```

Comment:
Issue: Because this is a v2 endpoint, could the tests for it please sit alongside the tests for my repair endpoint (in test/java/com/datastx/mgmtapi/resources/v2`?

Reply:
For unit tests (which I didn't write), that would be fine. This is an integration test though,

The integration tests have a bit of complexity with them as they are categorized as either "Isolated" or "Parallel". "Isolated" tests are ones that extend [BaseDockerIsolatedIntegrationTest](https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/src/test/java/com/datastax/mgmtapi/BaseDockerIsolatedIntegrationTest.java) and do so because they create a non-predictable Cassandra state after they run and the state is difficult to restore. For example the "DestructiveOpsIT" tests and the "LifecycleIT" tests fall into this category. The main difference between these tests is that the Isolated ones will stop the Docker container running Cassandra after each test and start a new one for the next test so that results are predictable. The non-isolated tests spin up a Docker container and run all tests against the same container without stopping it in between tests.

If that were all, I could simply create an IT test in the new `v2` test package and be done. Except that isn't all. The IT tests are now run in [parallel]()https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/.github/workflows/ci.yaml#L14 in GitHub CI to try to speed things up. So if we create a new IT test in the `v2` package, I'll have to add it there as well, but for all the tests matrices (3.11, 4.0, 4.1, DSE, etc). By putting this test specifically in the `NonDestructiveOpsIT` test class, it will already run in CI.

So a follow-up question to you:
Do you want to add this new IT test class in the `v2` package and add it to the CI test matrices?

Reply:
Actually, I might be able to do both. The way CI uses the test matrix is by specifying the test class name. So as long as I name the `v2` version of the Integration tests the same as one of the existing ones, I may not have to mess with the test matrix. Will test this locally and report back.

Reply:
>  Do you want to add this new IT test class in the v2 package and add it to the CI test matrices?

Uh, sorry, I didn't quite understand what was going on here with it being an e2e test against an actual running management API. I don't mind too much now I know that. e2e tests don't strictly need to follow the structure/packaging of the production code in our other projects.

Having said that, well, it is nice to have smaller test classes I guess, so maybe this change is for the good. 

Suggestion/nit: Do you think we could call it `NonDestructiveOpsResourcesV2IT.java` though, obviously the naming of these tests (`IT` instead of `E2E`) is already confusing me a bit.